src/util/file-filter.js

```diff
@@ -0,0 +1,87 @@
+/* @flow */
```

Comment:
Good call on moving this to its own file. You also need to move the tests out of [test.build.js](https://github.com/mozilla/web-ext/blob/master/tests/unit/test-cmd/test.build.js#L308). The new `ignoreFiles` needs test coverage too.